[CI] InferenceActionRequestTests testWriteTo_ForHasBeenReroutedChanges failing

**Build Scans:**
- [elasticsearch-intake #38803 / part2](https://gradle-enterprise.elastic.co/s/c7rjpvfrrk4aq)
- [elasticsearch-pull-request #141543 / part-2](https://gradle-enterprise.elastic.co/s/zrjf6oc3yu6ie)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:core:test" --tests "org.elasticsearch.xpack.core.inference.action.InferenceActionRequestTests.testWriteTo_ForHasBeenReroutedChanges" -Dtests.seed=DE56DDDD47A4FAA1 -Dtests.locale=lv-LV -Dtests.timezone=Australia/Yancowinna -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.core.inference.action.InferenceActionRequestTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testWriteTo_ForHasBeenReroutedChanges'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<InferenceAction.Request(taskType=text_embedding, inferenceEntityId=model, query=null, returnDocuments=null, topN=null, input=[input], taskSettings={}, inputType=unspecified, timeout=0s, context=InferenceContext[productUseCase=])> but was:<InferenceAction.Request(taskType=text_embedding, inferenceEntityId=model, query=null, returnDocuments=null, topN=null, input=[input], taskSettings={}, inputType=unspecified, timeout=30s, context=InferenceContext[productUseCase=])>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testWriteTo_ForHasBeenReroutedChanges
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testWriteTo_ForHasBeenReroutedChanges (0.5% fail rate in 367 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -505,9 +505,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
   method: test {csv-spec:fuse.fuseWithRowLinearAndMultiValueGroupColumn}
   issue: https://github.com/elastic/elasticsearch/issues/147988
-- class: org.elasticsearch.xpack.core.inference.action.InferenceActionRequestTests
-  method: testWriteTo_ForHasBeenReroutedChanges
-  issue: https://github.com/elastic/elasticsearch/issues/148000
 - class: org.elasticsearch.search.CCSDuelIT
   method: testPagination
   issue: https://github.com/elastic/elasticsearch/issues/148006

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/action/InferenceActionRequestTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/action/InferenceActionRequestTests.java
@@ -22,7 +22,6 @@ import org.elasticsearch.xpack.core.ml.AbstractBWCWireSerializationTestCase;
 
 import java.io.IOException;
 import java.util.List;
-import java.util.Map;
 
 import static org.elasticsearch.xpack.core.inference.action.BaseInferenceActionRequest.INFERENCE_REQUEST_PER_TASK_TIMEOUT_ADDED;
 import static org.elasticsearch.xpack.core.inference.action.BaseInferenceActionRequest.TIMEOUT_NOT_DETERMINED;
@@ -604,31 +603,5 @@ public class InferenceActionRequestTests extends AbstractBWCWireSerializationTes
             false,
             context
         );
-    }
-
-    public void testWriteTo_ForHasBeenReroutedChanges() throws IOException {
-        var instance = new InferenceAction.Request(
-            TaskType.TEXT_EMBEDDING,
-            "model",
-            null,
-            null,
-            null,
-            List.of("input"),
-            Map.of(),
-            InputType.UNSPECIFIED,
-            randomTimeValue(),
-            false
-        );
-        {
-            // From a version with rerouting removed
-            InferenceAction.Request deserializedInstance = copyWriteable(
-                instance,
-                getNamedWriteableRegistry(),
-                instanceReader(),
-                BaseInferenceActionRequest.INFERENCE_REQUEST_ADAPTIVE_RATE_LIMITING_REMOVED
-            );
-
-            assertEquals(instance, deserializedInstance);
-        }
     }
 }

